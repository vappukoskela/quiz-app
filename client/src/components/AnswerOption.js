import { Checkbox, ListItem, ListItemIcon, ListItemText } from "@material-ui/core";
import { withStyles } from '@material-ui/core/styles';
import { green } from '@material-ui/core/colors';

function AnswerOption(props) {
    const GreenCheckbox = withStyles({
        root: {
            color: green[400],
            '&$checked': {
                color: green[600],
            },
        },
        checked: {},
    })((props) => <Checkbox color="default" {...props} />);

    return (
        <ListItem key={props.value.id} role={undefined} dense >
            <ListItemIcon>
                {props.answersVisible ?
                    <GreenCheckbox
                        checked={props.value.correct}
                        edge="start"
                        tabIndex={-1}
                        disabled={true}
                    /> : null}
            </ListItemIcon>
            <ListItemIcon>
                <Checkbox
                    // TODO:
                    // onChange={(event) => dispatch({ type: "SELECT_TOGGLE", data: { newText: event.target.value, quizIndex: quiz, questionIndex: parentIndex, answerIndex: index } })}
                    checked={props.value.selected}
                    edge="start"
                    tabIndex={-1}
                />
            </ListItemIcon>
            <div><ListItemText id={props.index} primary={props.value.answer} /></div>
        </ListItem>
    )
}
export default AnswerOption;