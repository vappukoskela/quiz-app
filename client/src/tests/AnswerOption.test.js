import React from 'react';
import renderer from 'react-test-renderer';
import AnswerOption from '../components/AnswerOption.js';

// Komponentti joka vaatii propsit renderöityäkseen oikein!

// FAIL: ei propseja. Checkboxin checked-value on undefined - ei renderöidy!

// it('renders correctly', () => {
//     const tree = renderer.create(<AnswerOption />).toJSON();
//     expect(tree).toMatchSnapshot();
// });

it('renders correctly with props', () => {
    const value = {id: 0, correct: false, selected: false}
    const tree = renderer.create(<AnswerOption value={value} answersVisible={true} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('does not show answer when answers not visible', () => {
    const value = {id: 0, correct: false, selected: false}
    const tree = renderer.create(<AnswerOption value={value} answersVisible={false} />).toJSON();
    expect(tree).toMatchSnapshot();
});
