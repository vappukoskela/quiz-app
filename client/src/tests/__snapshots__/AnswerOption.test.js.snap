// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not show answer when answers not visible 1`] = `
<li
  className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemIcon-root"
  >
    <span
      aria-disabled={true}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root Component-root-7 PrivateSwitchBase-disabled-5 Mui-disabled Mui-disabled MuiIconButton-edgeStart Mui-disabled"
      hidden={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-6"
          data-indeterminate={false}
          disabled={true}
          onChange={[Function]}
          tabIndex={-1}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="MuiListItemIcon-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-6"
          data-indeterminate={false}
          onChange={[Function]}
          tabIndex={-1}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
  </div>
  <div>
    <div
      className="MuiListItemText-root MuiListItemText-dense"
    />
  </div>
</li>
`;

exports[`renders correctly with props 1`] = `
<li
  className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemIcon-root"
  >
    <span
      aria-disabled={true}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root Component-root-1 PrivateSwitchBase-disabled-5 Mui-disabled Mui-disabled MuiIconButton-edgeStart Mui-disabled"
      hidden={true}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-6"
          data-indeterminate={false}
          disabled={true}
          onChange={[Function]}
          tabIndex={-1}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="MuiListItemIcon-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-6"
          data-indeterminate={false}
          onChange={[Function]}
          tabIndex={-1}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div>
    <div
      className="MuiListItemText-root MuiListItemText-dense"
    />
  </div>
</li>
`;
